src/hook/useRegister.ts

```diff
@@ -0,0 +1,115 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { UserData } from '@/models/userModels';
+import { registerUser } from '@/services/authService';
```

Comment:
utilizar los nombres sin terminar en el tipo de que estos tienen
```suggestion
import { UserData } from '@/models/user';
import { registerUser } from '@/services/register';
```